x-pack/plugin/sql/sql-action/src/test/java/org/elasticsearch/xpack/sql/action/SqlRequestParsersTests.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.action;
+
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.xcontent.DeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.sql.proto.SqlTypedParamValue;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+
+public class SqlRequestParsersTests extends ESTestCase {
+
+    public void testUnknownFieldParsingErrors() throws IOException {
+        XContentParser parser = parser("{\"key\" : \"value\"}");
+        Exception e = expectThrows(IllegalArgumentException.class, () -> SqlClearCursorRequest.fromXContent(parser));
+        assertThat(e.getMessage(), containsString("unknown field [key]"));
+        
+        XContentParser parser2 = parser("{\"key\" : \"value\"}");
+        e = expectThrows(IllegalArgumentException.class, () -> SqlQueryRequest.fromXContent(parser2));
+        assertThat(e.getMessage(), containsString("unknown field [key]"));
+        
+        XContentParser parser3 = parser("{\"key\" : \"value\"}");
+        e = expectThrows(IllegalArgumentException.class, () -> SqlTranslateRequest.fromXContent(parser3));
+        assertThat(e.getMessage(), containsString("unknown field [key]"));
+    }
+    
+    public void testClearCursorRequestParser() throws IOException {
```

Comment:
Please also check for missing completely `client.id` and `mode`.
If you extract the parsing in a method then you could also unit test the method so we can avoid checking for all different types of requests.